x/mongo/driver/session/client_session.go

```diff
@@ -459,18 +469,21 @@ func (c *Client) ApplyCommand(desc description.Server) {
 			c.PinnedServer = &desc
 		}
 	} else if c.TransactionState == Committed || c.TransactionState == Aborted {
-		c.clearTransactionOpts()
 		c.TransactionState = None
+		return c.clearTransactionOpts()
```

Comment:
Wouldn't this have the effect of unpinning a connection after the first commit?

Reply:
If the application calls `commitTransaction` again, we'll set the `Committing` field to true so we'd exit in the first conditional of this function. We'd only get here if the transaction was committed/aborted and we're now starting a new transaction or non-transactional operation, which should be in line with the spec.